crates/matrix-sdk-ui/src/timeline/controller/state.rs

```diff
@@ -196,6 +194,21 @@ impl TimelineState {
         txn.commit();
     }
 
+    fn should_add_event_item(
+        timeline_focus: &TimelineFocusKind,
+        thread_root: &Option<OwnedEventId>,
+    ) -> bool {
+        match &timeline_focus {
+            TimelineFocusKind::Live => thread_root.is_none(),
+
+            TimelineFocusKind::Thread { root_event_id } => {
+                thread_root.as_ref().is_some_and(|r| r == root_event_id)
+            }
+
+            TimelineFocusKind::Event | TimelineFocusKind::PinnedEvents => false,
```

Comment:
This means that local echoes for aggregations won't apply to those focus kinds, which is likely wrong.

Reply:
errr, *I'm* wrong! aggregations are handled separately.